tools/test-npm.sh

```diff
@@ -17,6 +17,19 @@ rm -rf test-npm
 # make a copy of deps/npm to run the tests on
 cp -r deps/npm test-npm
 
+
+## a hopefully temporary hack to use the local $NODE binary when calling node(1)
+# make a link to $NODE as test-npm/node
+ln -fs $NODE node
+# get the absolute path we are in -- see: http://stackoverflow.com/a/4774063/1279026
+pushd `dirname $0` > /dev/null
+PWD=`pwd -P`
+popd > /dev/null
+# export that link as preferred in the $PATH
+export PATH=$PWD:$PATH
+##
```

Comment:
Looks like this fails on a more recent npm with these errors:

```
test/tap/lifecycle-path.js ............................ 2/4
  make sure the path is correct
  not ok exit code
    +++ found                                                          
    --- wanted                                                         
    -0                                                                 
    +1                                                                 
    compare: ===
    at:
      file: test/tap/lifecycle-path.js
      line: 55
      column: 7
    stack: |
      test/tap/lifecycle-path.js:55:7
      f (node_modules/once/once.js:17:25)
      ChildProcess.<anonymous> (test/common-tap.js:56:5)
      maybeClose (internal/child_process.js:764:16)
      Socket.<anonymous> (internal/child_process.js:319:11)
      Pipe._onclose (net.js:467:12)
  
  make sure the path is correct
  not ok should be equivalent
    +++ found                                                          
    --- wanted                                                         
     [                                                                 
    -  "{{ROOT}}/bin/node-gyp-bin"                                     
    -  "{{ROOT}}/test/tap/lifecycle-path/node_modules/.bin"            
    -  "/bin"                                                          
    -  "/usr/bin"                                                      
    +  "> ./node-bin/node print-path.js"                               
     ]                                                                 
    at:
      file: test/tap/lifecycle-path.js
      line: 74
      column: 7
    stack: |
      test/tap/lifecycle-path.js:74:7
      f (node_modules/once/once.js:17:25)
      ChildProcess.<anonymous> (test/common-tap.js:56:5)
      maybeClose (internal/child_process.js:764:16)
      Socket.<anonymous> (internal/child_process.js:319:11)
      Pipe._onclose (net.js:467:12)
```

From this test: https://github.com/npm/npm/blob/master/test/tap/lifecycle-path.js

@othiym23 is this worth the effort to hack this in before npm's tests work without a global node? It seems like this test is relying on the location of the node binary, is that something that can / should be fixed?